backend/lib/endpoints/auth.js

```diff
@@ -135,6 +136,31 @@ async function routes(app) {
       throw app.httpErrors.internalServerError();
     }
   });
+
+  app.post(
+    "/change-password",
+    { preHandler: [app.getServerToken], schema: changePasswordSchema },
+    async (req) => {
+      const { body, token } = req;
+      const { email } = body;
+
+      try {
+        const responseMessage = await Auth0.changePassword(token, email);
+        req.log.info(
+          `Change password email created successfully for email=${email}`,
+        );
+        return { email, responseMessage, token };
+      } catch (err) {
+        if (err.statusCode === 400) {
+          throw app.httpErrors.badRequest(err, `User ${email} does not exist`);
```

Comment:
what's the difference between "not exist" and "not found" for 404 here? I would think 400 is bad request, not sure if I understand

Reply:
The idea behind this validation came from the response messages from the `POST	/api/v2/tickets/password-change` API endpoint https://auth0.com/docs/api/management/v2#!/Tickets/post_password_change but the `/dbconnections/change_password` endpoint will actually accept any string, even if it's not a valid email address, and send that string an email. 

I tested the  `/dbconnections/change_password` endpoint and even an email string containing just `@` or `@mail.com` is valid and an email is sent. Granted those email addresses don't exist it's funny that the Auth0 endpoint does zero email validation and just returns "We've just sent you an email to reset your password." regardless of what the string input contains. 

Anyway, this error validation should not be here because the `/dbconnections/change_password` does not raise these errors. I will delete this conditional block. Thanks for pointing this out!

Reply:
Looks like this is resolved and I'm generally good with the PR with the changes you've made. 

The `/dbconnections/change_password` is intended to also be usable directly from a frontend (unlike `password-change` in Management API, which is for backend servers only) so for security reasons they wouldn't want to reveal if the email exists in the db or not.

Not sure for the reason to not validate the format however. That is something I think we should still do ourselves with `email: S.string().format("email"),`

Reply:
Actually looking at the other auth.js endpoints they aren't using .format("email"). Maybe an oversight or is there a reason @robinv85 ? If an oversight I'd make that as a separate low priority issue.

Reply:
@joshmorel validation of the API through schema's is definitely an important point that needs improvement
I've been working on adding some more validations for the update user endpoint but it definitely takes some effort to get it right, `fluent-schema` should just be syntactic sugar for JSON schema's, but AJV actually includes some custom formats on top of the JSON schema standards. We also have to make sure the backend and frontend have the same validations, I've jsut tested a regular expression directly from the AJV core and it wasn't even accepted as a valid regex, but it works on the frontend.

Reply:
Also, for the email validation on the frontend we have pretty advanced validations, like the domain extension is actually checked, but that might not work for other valid emails like e.g. an IP address which would probably be accepted by the standard email format of JSON schema, and it would be harder to add this domain extension validation on the backend in the JSON schema's as well.